### PR TITLE
Sign verify message flow v2

### PR DESCRIPTION
**Pull request by PeterZlodej**
[ch2489]

### CODE REVIEW

File: common/v2/features/SignAndVerifyMessage/SignMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+            </SignedMessage>
+          )}
+        </>
+      )}
+    </Content>
+  );
+}
+
+const mapStateToProps = (state: AppState) => ({
+  nodeLib: configNodesSelectors.getNodeLib(state),
```
Comment (vilsbole) [inline common/v2/features/SignAndVerifyMessage/SignMessage.tsx]: Select the network here.

File: common/v2/features/SignAndVerifyMessage/SignMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  sig: state.paritySigner.sig
+});
+
+const mapDispatchToProps = {
+  requestMessageSignature: paritySignerActions.requestMessageSignature
```
Comment (vilsbole) [inline common/v2/features/SignAndVerifyMessage/SignMessage.tsx]: This seems like a mistake to default to Parity.
Am I missing something?

File: common/v2/features/SignAndVerifyMessage/SignMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+          )}
+        </>
+      )}
+    </Content>
+  );
+}
+
+const mapStateToProps = (state: AppState) => ({
+  nodeLib: configNodesSelectors.getNodeLib(state),
+  sig: state.paritySigner.sig
```
Comment (vilsbole) [inline common/v2/features/SignAndVerifyMessage/SignMessage.tsx]: Is it intentional to default to Parity?!

File: common/v2/config/walletInfo.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
+import { KNOWLEDGE_BASE_URL as KB_URL } from 'v2/config';
+import {
+  SecureWalletName,
+  InsecureWalletName,
+  MiscWalletName,
+  WalletType,
+  WalletName
+} from 'v2/types';
+import { getWeb3ProviderInfo } from 'utils/web3';
```
[the remark this answers is not in the document]
Comment (PeterZlodej) [reply common/v2/config/walletInfo.ts]: Done

File: common/v2/config/walletInfo.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+// @ADD_ACCOUNT_TODO: Icons really belongs to the WalletButton or a WalletIcon
+// component.
+import LedgerSVG from 'common/assets/images/wallets/ledger.svg';
+import TrezorSVG from 'common/assets/images/wallets/trezor.svg';
+import SafeTSVG from 'common/assets/images/wallets/safe-t.png';
+import ParitySignerSVG from 'common/assets/images/wallets/parity-signer.svg';
+
+// STORIES serve the double purpose of generating the wallet options and
+// providing a declarative way to determine the flow for each wallet.
```
[the remark this answers is not in the document]
Comment (PeterZlodej) [reply common/v2/config/walletInfo.ts]: Removed

File: common/v2/components/WalletList/WalletList.tsx [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+interface Props {
+  wallets: any[];
+  showHeader?: boolean;
+  onSelect(name: WalletName): void;
+}
+
+export default class WalletList extends PureComponent<Props> {
+  public render() {
+    const { wallets, onSelect, showHeader } = this.props;
+    const validWallets = wallets.filter(w => !w.hideFromWalletList); // @TODO Filter here according to electronOnly
```
[the remark this answers is not in the document]
Comment (PeterZlodej) [reply common/v2/components/WalletList/WalletList.tsx]: Removed

File: common/v2/features/BroadcastTx/BroadcastTx.tsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+      this.setState({ inputError: translateRaw('BROADCAST_TX_INPUT_ERROR') });
+    }
+  };
+
+  protected handleChange = ({ currentTarget }: React.FormEvent<HTMLInputElement>) => {
+    const { value } = currentTarget;
+    this.setState({ userInput: value });
+
+    if (value === '') {
+      this.setState({ inputError: '' });
```
[the remark this answers is not in the document]
Comment (PeterZlodej) [reply common/v2/features/BroadcastTx/BroadcastTx.tsx]: Fixed

File: common/v2/features/SignAndVerifyMessage/SignMessage.tsx (hunk 1 above)
Comment (PeterZlodej) [reply common/v2/features/SignAndVerifyMessage/SignMessage.tsx]: Done, automatically selected web3 node.

File: common/v2/features/SignAndVerifyMessage/SignMessage.tsx (hunk 2 above)
Comment (PeterZlodej) [reply common/v2/features/SignAndVerifyMessage/SignMessage.tsx]: This is parity-specific implementation, we renamed the variables to make that more clear.

File: common/v2/features/SignAndVerifyMessage/SignMessage.tsx (hunk 3 above)
Comment (PeterZlodej) [reply common/v2/features/SignAndVerifyMessage/SignMessage.tsx]: This is parity-specific implementation, we renamed the variables to make that more clear.

**Comment by vilsbole** [comment]
lgtm;

**Pull request merged**